lib/src/api/trace/trace_state.dart

```diff
@@ -70,7 +70,7 @@ class TraceState {
     return (match != null) && (match.group(0) == value);
   }
 
-  String get(String key) => _state[key];
+  String? get(String key) => _state[key];
```

Comment:
should this return empty string if not found?
```suggestion
  String get(String key) => _state[key] ?? '';
```

Reply:
I'm not sure about this.  The current implementation is able to differentiate between "the state does not contain this key" and "the value for this key is an empty string".  I'm concerned that using a blank string here could allow blank values to leak into the trace state.

Reply:
https://opentelemetry.io/docs/specs/otel/trace/tracestate-handling

The grammar suggests that empty values are valid values. And based on this document, I do not think it is worth marking this APIs return value as `String?` because afaict `String` should be sufficient and in the case `String?` must change to `String` that is a breaking change. But going from `String` to `String?` is not a breaking change

Reply:
>The grammar suggests that empty values are valid values.

Yes, but that's not what I'm concerned about.  Do we want the library to start returning valid data for keys that don't exist?

Reply:
I would not recommend that instrumentation leveraging `tracestate` define null and empty string as two distinct states. I would consider this bad practice.

But this is just my opinion. Perhaps there's a valid reason for treating the two as distinct states. There isn't, but someone could make a case for it. At which point we could add `?` to the API and that change is backwards compatible.

Here's some pseudo code that pretends to use trace state that can distinguish between null and empty value:

```dart
const value = ctx.get('my-key')

if (value == null) {
// initialize value to something non-empty
} else if (value == "") {
// this is a special case that implies something special?
// or more likely we treat it the same as null and initialize the value to something non-empty
} else {
// value is non-null and non-empty
}
```